add: `rofi`

itsdonnix opened the pull request:
## Progress 

- [x] Edit packagelist
- [x] Add initial pacscript
- [x] Add maintainer to pacscript

itsdonnix committed:
add: `rofi`

Signed-off-by: Don Alfons <donnisnoni@outlook.com>

itsdonnix committed:
fix: correct rofi package url

- url was missing single quotes.
- added single quotes for consistency.

Signed-off-by: Don Alfons <donnisnoni@outlook.com>

itsdonnix committed:
fix: rofi check target exit code

- the check target should not allow errors
- this ensures proper build process handling

Signed-off-by: Don Alfons <donnisnoni@outlook.com>

itsdonnix committed:
chore: remove clean function

- remove the clean function
- not needed in the script

Signed-off-by: Don Alfons <donnisnoni@outlook.com>

Elsie19 commented on packages/rofi/rofi.pacscript:
```diff
@@ -0,0 +1,75 @@
+
+package() {
+  cd "${pkgname}~${pkgver}"
+  make install
```
```suggestion
  make DESTDIR="${pkgdir}" install
```

Elsie19 commented on packages/rofi/rofi.pacscript:
```diff
@@ -0,0 +1,75 @@
+  cd "${pkgname}~${pkgver}"
+  make install
+
+  rm -rf "${pkgdir}/usr/include/rofi"
+  rm -f "${pkgdir}/usr/lib/pkgconfig/rofi.pc"
```
Why are you removing these?

itsdonnix replied:
I think because I don't know how to deliver two packages: `rofi` and `rofi-dev` 😁

itsdonnix replied:
Alright, will try to split the package

itsdonnix committed:
fix: set correct install prefix and DESTDIR

- Use /usr as prefix instead of embedding pkgdir
- Set DESTDIR during install to stage files properly

Signed-off-by: Don Alfons <donnisnoni@outlook.com>

itsdonnix commented:
I notice this one: #7023 

Maybe in the future if it has been added, should be `rofi-wayland` or something?

itsdonnix committed:
feat: split into runtime and dev subpackages

- Adds `rofi-dev` for development headers and .pc files
- Updates package metadata and dependencies
- Moves dev files out of main `rofi` package

Signed-off-by: Don Alfons <donnisnoni@outlook.com>
